### PR TITLE
docs(secret-manager): add secret protection concept

### PR DESCRIPTION
**Pull request by nerda-codes**

### CODE REVIEW

File: identity-and-access-management/secret-manager/concepts.mdx [hunk 1]
```diff
@@ -74,6 +74,15 @@ Scaleway's Secret Manager can be managed using [APIs](https://www.scaleway.com/e
 
 <Concept>
 
+## Secret protection
+
+Secret protection is a feature that allows users with the `SecretManagerFullAccess`, `SecretManagerSecretWrite`, and `AllProductsFullAccess` [permission sets](/identity-and-access-management/iam/reference-content/permission-sets/), to enable it on your secrets, so that even though they can be read and modified, they **cannot be deleted**.
```
Comment (SamyOubouaziz) [inline identity-and-access-management/secret-manager/concepts.mdx:79]: Maybe start with the feature description, then premission sets? "Secret protection is a feature that prevents your secrets from being deleted. users with the blablabla permission sets can enable it...."

File: identity-and-access-management/secret-manager/concepts.mdx [hunk 2]
```diff
@@ -62,7 +62,7 @@ Secrets are logical containers made up of zero or more immutable versions, that
 
 ## Secret Manager
 
-Secret Manager is a solution that provides a centralized location where your secrets can be stored and accessed. Its primary purpose is to securely store secrets, thus eliminating the need for developers to hard code sensitive information in applications or configuration files.
+Secret Manager is a solution that provides a centralized location where your secrets can be stored and accessed. Its primary purpose is to securely store secrets, thus eliminating the need for developers to hardcode sensitive information in applications or configuration files.
```
Comment (bene2k1) [inline identity-and-access-management/secret-manager/concepts.mdx:65]: Are you sure about this? According to the internet it is written as two words https://en.wiktionary.org/wiki/hard_code

File: identity-and-access-management/secret-manager/concepts.mdx (hunk 1 above)
Comment (crlptl) [reply identity-and-access-management/secret-manager/concepts.mdx:79]: I plussoie Samy.

**Review by crlptl — COMMENTED** [review]
Apart from both Samy and Bene's comment, LGTM

File: identity-and-access-management/secret-manager/concepts.mdx (hunk 1 above)
Comment (nerda-codes) [inline identity-and-access-management/secret-manager/concepts.mdx:79]:
```suggestion
Secret protection is a feature that prevents the deletion of your secrets.  Users holding the `SecretManagerFullAccess`, `SecretManagerSecretWrite`, and `AllProductsFullAccess` [permission sets](/identity-and-access-management/iam/reference-content/permission-sets/) can enable it, ensuring that although your secrets can be accessed and modified, they remain **impossible to delete**.
```

File: identity-and-access-management/secret-manager/concepts.mdx (hunk 2 above)
Comment (nerda-codes) [inline identity-and-access-management/secret-manager/concepts.mdx:65]:
```suggestion
Secret Manager is a solution that provides a centralized location where your secrets can be stored and accessed. Its primary purpose is to securely store secrets, thus eliminating the need for developers to hard code sensitive information in applications or configuration files.
```

**Review by bene2k1 — APPROVED** [review]

**Review by RoRoJ — APPROVED** [review]

**Pull request merged**